llrt_core/src/modules/http/headers.rs

```diff
@@ -28,49 +28,89 @@ impl Headers {
         if let Some(init) = init.into_inner() {
             if init.is_array() {
                 let array = init.into_array().unwrap();
-                let headers = array_to_btree_map(&ctx, array)?;
-                return Ok(Self::from_map(headers));
+                let headers = Self::array_to_headers(array)?;
+                return Ok(Self { headers });
             } else if init.is_object() {
                 return Self::from_value(&ctx, init);
             }
         }
-
         Ok(Self {
-            headers: BTreeMap::default(),
+            headers: Vec::new(),
         })
     }
 
     pub fn append(&mut self, key: String, value: String) {
         let key = key.to_lowercase();
-
-        self.headers
-            .entry(key)
-            .and_modify(|header| *header = format!("{}, {}", header, &value))
-            .or_insert_with(|| value);
+        if key == HEADERS_KEY_SET_COOKIE {
+            return self.headers.push((key, value));
+        }
+        if let Some((_, existing_value)) = self.headers.iter_mut().find(|(k, _)| k == &key) {
+            match key.as_str() {
+                HEADERS_KEY_COOKIE => existing_value.push_str("; "),
+                _ => existing_value.push_str(", "),
+            }
+            existing_value.push_str(&value);
+        } else {
+            self.headers.push((key, value));
+        }
     }
 
-    pub fn get(&mut self, key: String) -> Option<String> {
-        self.headers.get(&key.to_lowercase()).map(|v| v.to_owned())
+    pub fn get(&self, key: String) -> Option<String> {
+        let key = key.to_lowercase();
+        let mut result = String::new();
+        for (k, v) in &self.headers {
+            if k == &key {
```

Comment:
This function supports the case where there are multiple `set-cookies`.

Reply:
I think we can make this a bit more efficient like this:
 ```rust
pub fn get(&self, key: String) -> Option<String> {
       let key = key.to_lowercase();
       if key ==  HEADERS_KEY_SET_COOKIE {
           let result: Vec<String> = pairs.iter()
            .filter_map(|(k, v)| {
                if k == &key {
                    Some(v.clone())
                } else {
                    None
                }
            })
            .collect();
        
            return if result.is_empty() {
                None
            } else {
                Some(result.join("; "))
            }
       }

       &self.headers.iter().find(|(k,_)|k == &key).map(|(_,v)|v.clone())
 
 }

```

This will only join and do branch checks if key is `HEADERS_KEY_SET_COOKIE`. It also saves on allocations as we don't grow a string for each key.

Reply:
I will incorporate this improvement, but will make a few changes to the non `set-cookie` as it is easier to read if it is handled earlier.

Reply:
Thanks for pointing this out. I applied it immediately.